model/role.go

```diff
@@ -88,6 +95,36 @@ func (r *Role) Patch(patch *RolePatch) {
 	}
 }
 
+// MergeHigherScopedPermissions is meant to be invoked on a channel scheme's role and merges the higher-scoped
+// channel role's permissions.
+func (r *Role) MergeHigherScopedPermissions(higherScopedPermissions *RolePermissions) {
+	mergedPermissions := []string{}
+
+	for _, cp := range ALL_PERMISSIONS {
+		if cp.Scope != PERMISSION_SCOPE_CHANNEL {
+			continue
+		}
+
+		if higherScopedPermissions.RoleID == CHANNEL_ADMIN_ROLE_ID && slices.IncludesString(higherScopedPermissions.Permissions, cp.Id) {
+			mergedPermissions = append(mergedPermissions, cp.Id)
+			continue
+		}
+
+		if _, ok := ModeratedPermissions[cp.Id]; ok {
+			if slices.IncludesString(r.Permissions, cp.Id) && slices.IncludesString(higherScopedPermissions.Permissions, cp.Id) {
+				mergedPermissions = append(mergedPermissions, cp.Id)
+			}
+			continue
```

Comment:
Same here - I think a helper method on Role that converts the an array of permissions into a map would help

Reply:
Done.